docs/features/interceptors.md

```diff
@@ -0,0 +1,168 @@
+
+#### Attribute creation
+
+The goal of the above decisions is to make it so that when source generators are filling in `[InterceptsLocation(...)]`, they simply need to read `nameSyntax.SyntaxTree.FilePath` and `nameSyntax.GetLineSpan().Span.Start` for the exact file path and position information they need to use.
+
+We add the public API `public static InterceptableLocation? GetInterceptableLocation(this InvocationExpressionSyntax syntax)`. The API and the new return type are guarded by `[RequiresPreviewFeatures]`, requiring generators which use the APIs to opt-in. If the feature is significantly revised or withdrawn, use of `[RequiresPreviewFeatures]` gives us the latitude to make breaking changes to these public APIs or to remove them.
+
+We should provide samples of recommended coding patterns for generator authors.
+
+### Non-invocation method usages
+
+Conversion to delegate type, address-of, etc. usages of methods cannot be intercepted.
+
+Interception can only occur for calls to ordinary member methods--not constructors, delegates, properties, local functions, etc.
+
+### Arity
+
+Interceptors cannot have type parameters or be declared in generic types at any level of nesting.
+
+### Signature matching
+
+When a call is intercepted, the interceptor and interceptable methods must meet the signature matching requirements detailed below:
+- When an interceptable instance method is compared to a classic extension method, we use the extension method in reduced form for comparison. The extension method parameter with the 'this' modifier is compared to the instance method 'this' parameter.
+- The returns and parameters, including the 'this' parameter, must have the same ref kinds and types, except that reference types with oblivious nullability can match either annotated or unannotated reference types.
+- Method names and parameter names are not required to match.
+- Parameter default values are not required to match. When intercepting, default values on the interceptor method are ignored.
+- 'scoped' modifiers and `[UnscopedRefAttribute]` must be equivalent.
+
+Arity does not need to match between intercepted and interceptor methods. In other words, it is permitted to intercept a generic method with a non-generic interceptor.
+
+### Conflicting interceptors
+
+If more than one interceptor refers to the same location, it is a compile-time error.
+
+If an `[InterceptsLocation]` attribute is found in the compilation which does not refer to the location of an explicit method call, it is a compile-time error.
+
+### Interceptor accessibility
+
+An interceptor must be accessible at the location where interception is occurring.
```

Comment:
Where do we check that?

Reply:
Added a note to the doc. In a subsequent PR I'd like to address this as well as scoped differences.